### ISSUE TITLE
Truncated Backpropagation Through Time

### ISSUE THREAD

**Issue by svnv-svsv-jm** [jdb78/pytorch-forecasting#1581]
Is Truncated Backpropagation Through Time supported?

I've found no documentation or example about it.

**Comment by benHeid** [comment]
It seems that it is not supported. At least I haven't found it in `training_step` nor in the `step` method implemented. I think this could be a nice add on for autoregressive or recurrent networks.

For implementing it, I suppose it would require to add for the recurrent models a new `step` method or add a hook in the `step` method of the `BaseModel` for the calculation of the loss.

**Comment by svnv-svsv-jm** [comment]
PyTorch Lightning had this feature, then they removed it. We could take inspiration from their old code?
